File: codegen/results/v002/gemma4_26b_think/20260411_184609/typescript/004_json_filter/solution.ts
import * as fs from 'fs';

/**
 * Interface representing the structure of the input data objects.
 */
interface DataRecord {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

/**
 * Main function to solve the task.
 * 1. Reads input from 'input/data.json'.
 * 2. Filters records where 'active' is true and 'age' is 30 or older.
 * 3. Sorts the resulting records by 'name' in ascending order.
 * 4. Outputs the resulting JSON array to stdout.
 */
function main() {
  const inputPath = 'input/data.json';

  try {
    // Verify the existence of the file to avoid throwing unhandled exceptions
    if (fs.existsSync(inputPath)) {
      // Read the file content as a UTF-8 string
      const fileContent = fs.readFileSync(inputPath, 'utf8');
      
      // Parse the JSON string into an array of DataRecord objects
      const records: DataRecord[] = JSON.parse(fileContent);

      // Filter the records based on the criteria:
      // - active must be true
      // - age must be 30 or greater
      const filteredRecords = records.filter((record) => {
        return record.active === true && record.age >= 30;
      });

      // Sort the filtered records by name in ascending order using localeCompare
      // localeCompare is used for robust string comparison
      filteredRecords.sort((a, b) => a.name.localeCompare(b.name));

      // Convert the resulting array back to a JSON string and write to stdout.
      // We do not use indentation (pretty-printing) to ensure the output 
      // is a clean, single-line JSON array unless required.
      process.stdout.write(JSON.stringify(filteredRecords));
    }
  } catch (error) {
    // The requirement specifies: "Do not output anything else."
    // Therefore, we catch errors and prevent error messages from being sent to stdout.
  }
}

// Execute the program
main();

File: codegen/results/v002/gemma4_26b_think/20260411_184609/go/004_json_filter/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"sort"
)

// Person represents the structure of the objects in the JSON input.
type Person struct {
	Name   string  `json:"name"`
	Age    int     `json:"age"`
	Active bool    `json:"active"`
	Score  float64 `json:"score"`
}

func main() {
	// Define the path to the input file.
	filePath := "input/data.json"

	// Read the content of the JSON file.
	data, err := os.ReadFile(filePath)
	if err != nil {
		// If there is an error reading the file, we exit silently 
		// to adhere to the "Do not output anything else" requirement.
		return
	}

	// Unmarshal the JSON array into a slice of Person structs.
	var people []Person
	err = json.Unmarshal(data, &people)
	if err != nil {
		return
	}

	// Filter the slice based on the criteria:
	// active is true AND age is 30 or older.
	var filtered []Person
	for _, p := range people {
		if p.Active && p.Age >= 30 {
			filtered = append(filtered, p)
		}
	}

	// Sort the filtered slice by the "name" field in ascending order.
	sort.Slice(filtered, func(i, j int) bool {
		return filtered[i].Name < filtered[j].Name
	})

	// Marshal the filtered and sorted slice back into a JSON array.
	// Using MarshalIndent to provide a clean, readable output format.
	output, err := json.MarshalIndent(filtered, "", "  ")
	if err != nil {
		return
	}

	// Output the resulting JSON array to stdout.
	fmt.Println(string(output))
}